### PR TITLE
Update `Accordion::Item` border radius

### PR DESCRIPTION
**Pull request by jorytindall**
### :pushpin: Summary

If merged, this PR updates the border radius of the `Accordion::Item` small variant to 3px, matching Figma. 

### :hammer_and_wrench: Detailed description

However, a more systematic approach might be to scale the border radius for _all_ items corresponding with the size ([as outlined in this thread and examples](https://www.figma.com/design/iweq3r2Pi8xiJfD9e6lOhF?node-id=36870-71031#977041939)):
- Small, 3px (requires change in code)
- Medium, 5px (requires change in Figma and code)
- Large, 6px (requires no change)

Are there any thoughts around this topic?

### :link: External links

Jira ticket: [HDS-3952](https://hashicorp.atlassian.net/browse/HDS-3952)
Figma file: [Thread](https://www.figma.com/design/iweq3r2Pi8xiJfD9e6lOhF?node-id=37555-77311#964785115)

***

### 👀 Component checklist

- [ ] Percy was checked for any visual regression
- [ ] A changelog entry was added via [Changesets](https://github.com/changesets/changesets) if needed (see [templates here](https://hashicorp.atlassian.net/wiki/spaces/HDS/pages/3243114706/Changelog+authoring+best+practices#Templates))

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

[HDS-3952]: https://hashicorp.atlassian.net/browse/HDS-3952?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/components/src/styles/components/accordion.scss [hunk 1]
```diff
@@ -51,6 +52,7 @@
   --hds-accordion-item-content-padding-left: 12px;
   --hds-accordion-item-gap: 12px;
   --hds-accordion-item-icon-size: 20px;
+  --hds-accordion-item-border-radius: 6px;
```
Comment (LilithJames-HDS) [inline packages/components/src/styles/components/accordion.scss:55]: Should this one be 5px rather than 6px?

Comment (jorytindall) [reply packages/components/src/styles/components/accordion.scss:55]: This doesn't currently reflect what I proposed above, but rather reflects what is currently in Figma.

**Comment by KristinLBradley** [comment]
**Related Slack thread discussion:** https://hashicorp.slack.com/archives/C025N5V4PFZ/p1728407881500429

File: packages/components/src/styles/components/accordion.scss (hunk 1 above)
Comment (LilithJames-HDS) [reply packages/components/src/styles/components/accordion.scss:55]: Ah I see, I misread the description.

**Comment by jorytindall** [comment]
I'm going to close this as the consensus is to not move in this direction and update the Figma component instead.

**Pull request closed without merging**